qhana_plugin_registry/tasks/plugin_filter.py

```diff
@@ -37,34 +37,39 @@ def evaluate_plugin_filter(filter_string: str) -> list[RAMP]:
     """
     # TODO: store filter as dict in DB to avoid parsing
     plugin_filter: dict = json.loads(filter_string)
-    plugins = DB.session.query(RAMP).all()
+    # TODO: filter plugins in batches
+    plugin_mapping = {plugin.id: plugin for plugin in DB.session.query(RAMP).all()}
 
     def get_plugins_from_filter(filter_dict: dict) -> list[RAMP]:
         # TODO: add match for plugin description
         match filter_dict:
             case {"and": filter_expr}:
-                return set.intersection(*map(set, [get_plugins_from_filter(f) for f in filter_expr]))
+                return set.intersection(
+                    *(get_plugins_from_filter(f) for f in filter_expr)
+                )
             case {"or": filter_expr}:
-                return set.union(*map(set, [get_plugins_from_filter(f) for f in filter_expr]))
+                return set.union(*(get_plugins_from_filter(f) for f in filter_expr))
             case {"not": filter_expr}:
-                return set.difference(set(plugins), set(get_plugins_from_filter(filter_expr)))
+                return set(plugin_mapping.keys()) - get_plugins_from_filter(filter_expr)
```

Comment:
If I remember right you can directly use the keys view of a dict for set operations:
```suggestion
                return plugin_mapping.keys() - get_plugins_from_filter(filter_expr)
```